### PR TITLE
halium: fix slot suffix missing on syspart for some devices

### PR DESCRIPTION
**Pull request by kcxt**
Some A/B devices (OnePlus 6) have the systempart cmdline but don't
specify a slot suffix in order to avoid requiring that ubports be flashed to a particular slot, leading to the initramfs attempting to mount a partition that doesn't exist.

We check if there is a suffix specified and append `$ab_slot_suffix` otherwise. This shouldn't affect non-A/B devices as `$ab_slot_suffix` will be empty.

### CODE REVIEW

**Comment by kcxt** [comment]
Hi is there anything else I can do to get this merged?

File: scripts/halium
Comment (UniversalSuperBox) [inline scripts/halium]: `export`-ing from this script seems risky. It could cause this environment to become part of the userspace init. Is that what you are trying to achieve?

**Review by UniversalSuperBox — CHANGES_REQUESTED** [review]

File: scripts/halium
Comment (kcxt) [reply scripts/halium]: Hi, no that isn't necessarily what I want. I didn't consider that at all to be honest.

It seems like it would make more sense to make that variable accessible from everywhere in ramdisk init, rather than to repeat the command. But for this PR I only need to be able to access the slot suffix in the `mountroot` function, what do you think is the best way to approach this?

Perhaps it would make the most sense to just fetch it from cmdline again.

**Comment by kcxt** [comment]
Oh pants, I somehow completely fudged this PR, `ab_slot_suffix` doesn't even get set until AFTER mounting `$_syspart`... I could've sworn it worked for me, I must've mucked something up between testing and committing.... I'll amend and push again to fix.

File: scripts/halium
Comment (kcxt) [reply scripts/halium]: I've decided to instead just fetch the variable again from cmdline where it's needed.

**Review by NotKit — APPROVED** [review]

**Comment by Vince1171** [comment]
@UniversalSuperBox I think this can be merged (waiting for your approval)

**Comment by UniversalSuperBox** [comment]
Unfortunately, this conflicts with the merge of https://github.com/Halium/initramfs-tools-halium/pull/22. Could you merge the changes to ensure the final script is correct?

**Comment by Vince1171** [comment]
@fredldotme can you try/resolve this? I'm not really used to A/B devices

**Comment by kcxt** [comment]
@UniversalSuperBox @Vince1171 fixed! #22 removed the call to `identify_boot_mode` which confused git. I've rebased and resolved the conflict.

**Comment by Vince1171** [comment]
I unfortunately don't have another A/B device that the op6T so my testing wont be very extensive.

**Comment by kcxt** [comment]
Bump, I don't really want to be responsible for this PR for much longer if possible!

**Comment by fredldotme** [comment]
I'd be okay with merging it, but I don't have a device (or build setup, or time) to test this properly with.
In general it's a fine change, but I would hate to get slapped for merging it without tests.

Could we find someone with a halium-boot device (non-Jumpercable) with A/B to test this against, as well as testing it against the VollaPhone to ensure it doesn't break non-A/B (just to make sure, better safe than sorry)?

**Comment by Flohack74** [comment]
Should we target this for OTA-20?

**Comment by amartinz** [comment]
This change was carried around in ramdisk overlays already for quite some time.
I am sorry for just seeing this PR right now. I should have really checked before opening a new PR.

https://github.com/Halium/initramfs-tools-halium/pull/30

Essentially the same change got merged for the "dynamic partition" variant, which i am using for H10.

**Comment by kcxt** [comment]
Thanks @amartinz, I'm glad we could at least get this feature implemented at last.

**Pull request closed without merging**
